### PR TITLE
Mappet 0.3

### PR DESCRIPTION
**Pull request by PyXiion**
- Ингридиенты не показываются, если их нет.
- Добавлен метод isOnGround для IScriptEntity
- Добавлена поддержка обычных и стрелочных функций в mappet.createUI
  
- Добавлено событие взаимодействия игрока с сущностью
- Подсветка строк с форматированием  
Screenshot_20220807_152537
- Скрипты теперь создаются по умолчанию с параметром unique
- Добавлена возможность создавать тип ScriptVector `new Vector3(1, 2, 3);`
- Добавлены Java.import и Java.importAs
  

---

- Ingredients are not shown if there are none.
- Added isOnGround method for IScriptEntity
- Added support for embed and arrow functions in mappet.createUI
  Screenshot_20220808_003707
- Added a player interaction event with an entity
- Highlighting strings with formatting
Screenshot_20220807_152537
- Scripts are now created by default with the unique parameter
- Added the ability to create the ScriptVector type `new Vector3(1, 2, 3);`
- Added Java.import & Java.importAs

### CODE REVIEW

**Comment by Chryfi** [comment]
Why did you delete all of the gradle tasks?

File: build.gradle [hunk 1]
```diff
@@ -106,133 +165,21 @@ task scriptingJavadocsJson(type: Exec) {
 
 tasks.processResources.dependsOn(scriptingJavadocsJson)
 
-/* Comparing keys */
-task compareLangFiles {
```
Comment (Chryfi) [inline build.gradle:110]: This is not acceptable. How did you manage to build the lang files without these tasks?

File: build.gradle [hunk 2]
```diff
@@ -83,14 +150,6 @@ artifacts {
     archives sourcesJar
 }
 
-task scriptingJavadocs(type: Javadoc) {
```
Comment (Chryfi) [inline build.gradle:86]: I dont know exactly what this task does, McHorse can tell better, but I am 99.9% sure this is not supposed to be removed.

File: build.gradle [hunk 3]
```diff
@@ -1,67 +1,134 @@
...
+            }
+        }
+    }
 }
 
 processResources
 {
+    duplicatesStrategy = DuplicatesStrategy.INCLUDE
     inputs.property "version", project.version
-    inputs.property "mcversion", project.minecraft.version
```
Comment (Chryfi) [inline build.gradle:55]: Why did you remove the variable? Copy paste of constants is not good.

File: build.gradle [hunk 4]
```diff
@@ -1,67 +1,134 @@
...
+    // These dependencies get remapped to your current MCP mappings
+    // deobf 'com.mod-buildcraft:buildcraft:6.0.8:dev'
+
+    // For more info...
+    // http://www.gradle.org/docs/current/userguide/artifact_dependencies_tutorial.html
+    // http://www.gradle.org/docs/current/userguide/dependency_management.html
+
+    compileOnly files("run/libs/blockbuster-${blockbuster}-1.12.2-dev.jar")
+    compileOnly files("run/libs/metamorph-${metamorph}-1.12.2-dev.jar")
+    compileOnly files("run/libs/mclib-${mclib}-1.12.2-dev.jar")
```
Comment (Chryfi) [inline build.gradle:88]: Use the ${project.minecraft.version} variable, there was no need to change this.

File: build.gradle [hunk 5]
```diff
@@ -1,67 +1,134 @@
...
-    version = "1.12.2-14.23.3.2655"
-    runDir = "run"
-    
-    mappings = "snapshot_20171003"
-    
-    clientJvmArgs = ["-Xmx1G"]
-    serverJvmArgs = ["-Xmx1G"]
-
-    replace "%VERSION%", project.version
-    replace "%MCLIB%", project.mclib
```
Comment (Chryfi) [inline build.gradle:41]: Removing this is not acceptable. This is required for proper building, why did you remove this?

**Review by Chryfi — CHANGES_REQUESTED** [review]
I don't understand why all of these changes in build.gradle were needed? Please don't just change dependency versions and remove components without any need for removal.

File: build.gradle (hunk 1 above)
Comment (Chryfi) [reply build.gradle:110]: I see now that mappet, unlike most of the other McHorse mods, does not have yaml files, I misunderstood the purpose of these tasks... Looks like they are not needed for building lang files. Nonetheless, these tasks should not be removed. There was no need for removal. Please don't just remove things from the existing working system.

**Comment by PyXiion** [comment]
Okay, sorry, I'll try to improve.

File: src/main/java/mchorse/mappet/api/scripts/code/entities/ScriptEntity.java [hunk 6]
```diff
@@ -308,7 +313,11 @@ public IScriptEntity getTarget()
     @Override
     public void setTarget(IScriptEntity entity)
     {
-        if (this.entity instanceof EntityLiving && entity != null && entity.isLivingBase())
+        if (entity == null)
+        {
+            ((EntityLiving) this.entity).setAttackTarget(null);
```
Comment (Chryfi) [inline src/main/java/mchorse/mappet/api/scripts/code/entities/ScriptEntity.java:318]: Possible ClassCastException. Before it was checked if `this.entity instanceof EntityLiving`, now you directly cast to that class although it is not certain what type `this.entity` is.

**Review by Chryfi — CHANGES_REQUESTED** [review]

File: src/main/java/mchorse/mappet/api/scripts/code/entities/ScriptEntity.java (hunk 6 above)
Comment (PyXiion) [reply src/main/java/mchorse/mappet/api/scripts/code/entities/ScriptEntity.java:318]: Thanks :sweat_smile:

Comment (Chryfi) [reply src/main/java/mchorse/mappet/api/scripts/code/entities/ScriptEntity.java]: No problem 👍
